plugins/net_plugin/net_plugin.cpp

```diff
@@ -3685,6 +3790,68 @@ namespace eosio {
       cc.process_vote_message(connection_id, msg);
    }
 
+   // called from connection strand
+   void connection::handle_message( const block_nack_message& msg ) {
+      auto block_num = block_header::num_from_id(msg.id);
+
+      peer_elog(this, "received block nack #${bn}:${id}, consecutive ${c}", ("bn", block_num)("id", msg.id)("c", consecutive_blocks_nacks));
```

Comment:
Because we just received one. Also probably move after the next `if (block_num == 0) {`.
Actually better move this log message to right after we actually increment `consecutive_blocks_nacks`.

```suggestion
      peer_elog(this, "received block nack #${bn}:${id}, consecutive ${c}", ("bn", block_num)("id", msg.id)("c", consecutive_blocks_nacks+1));
```

Reply:
That doesn't hold if `block_num` is `0`. That should be a `peer_dlog` missed changing that back after testing. I moved it to after `consecutive_blocks_nacks` is updated.